opwen_email_server/integration/celery.py

```diff
@@ -101,6 +103,18 @@ def send(resource_id: str) -> None:
     action(resource_id)
 
 
+@celery.task(ignore_result=True)
+def process_echo_service_email(resource_id: str) -> None:
```

Comment:
I was actually thinking about the celery aspect because it still does require some digging through the codebase. This is definitely a great solution! It would be very easy to extend in the future. All one would have to do is add a new formatting class then add it to the registry! The dynamically registering registry would even remove the need for manually adding something to the registry!